File: Python Scala Spark/SparkPool/01.scala
// Lab - Spark Pool - Starting out with Notebooks

// Spark - Resilient Distributed Dataset
// We are creating a simple array of values
// Here val is used to represent an immutable data set
// If you want a mutable data set then you need to use the var construct

val data = Array(1, 2, 3, 4, 5)

// The parallelize method of the Spark context will create an RDD
val dist = sc.parallelize(data)

// To get the count of values in the RDD
dist.count()

// If you want to get the elements of the RDD
dist.collect()

File: Python Scala Spark/SparkPool/01.py

# Lab - Spark Pool - Spark DataFrames
# Here we are creating a data frame

courses = [(1,'AZ-900',10.99),(2,'DP-203',11.99) ,(3,'AZ-104',12.99)]
df = spark.createDataFrame(courses, ['Id', 'Name', 'Price'])
df.show()

# Then here we are descending the rows based on the value in the price column

from pyspark.sql.functions import desc
from pyspark.sql.functions import col

# Here col is a class that is used to represent a new column that will be constructed based on the input column data
# You can use different variations here , col("columnname") or $"columnname", df("columnname")

sorteddf=df.sort(col("Price").desc())
sorteddf.show()